docs/DesignDocs/InitTimeConfigurationInterface.md

```diff
@@ -0,0 +1,271 @@
+ * This function is only available in the enclave.
+ *
+ * @param[in] format_id The format ID of the evidence to be generated.
+ * @param[in] flags A bit-wise parameter. Currently there is one bit
+ * defined: OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID. If this bit is set,
+ * the evidence and endorsements will be wrapped with a header containing
+ * the format ID.
+ * @param[in] runtime_custom_claims_buffer The optional runtime custom claims
+ * buffer. When provided, the content of the buffer is included in the
+ * evidence_buffer, with integrity protection. Depending on the underlining TEE
+ * and the plugin implementation, the content might or might not be further
+ * encrypted.
+ * @param[in] runtime_custom_claims_buffer_size The number of bytes in the
+ * runtime custom claims buffer.
+ * @param[in] inittime_custom_claims_buffer The optional inittime custom claims
+ * buffer. When provided, the content of the buffer is included in the
+ * evidence_buffer as plaintext. The integrity protection of the content is
+ * provided by the underlining TEE and the enclave SW outside the plugin.
+ * @param[in] inittime_custom_claims_buffer_size The number of bytes in the
+ * inittime custom claims buffer.
+ * @param[in] optional_parameters The optional format-specific input parameters.
+ * @param[in] optional_parameters_size The size of optional_parameters in bytes.
+ * @param[out] evidence_buffer An output pointer that will be assigned the
+ * address of the dynamically allocated evidence buffer.
+ * @param[out] evidence_buffer_size A pointer that points to the size of the
+ * evidence buffer in bytes.
+ * @param[out] endorsements_buffer If not NULL, an output pointer that will be
+ * assigned the address of the dynamically allocated endorsements buffer.
+ * @param[out] endorsements_buffer_size A pointer that points to the size of the
+ * endorsements buffer in bytes.
+ * @retval OE_OK The function succeeded.
+ * @retval OE_INVALID_PARAMETER At least one of the parameters is invalid.
+ * @retval OE_NOT_FOUND The input evidence format id is not supported.
+ * @retval other appropriate error code.
+ */
+oe_result_t oe_get_evidence(
+    const oe_uuid_t* format_id,
+    uint32_t flags,
+    const void* runtime_custom_claims_buffer,
+    size_t runtime_custom_claims_buffer_size,
+    const oe_inittime_custom_claim_buffer_t* inittime_custom_claims_buffer,
+    size_t inittime_custom_claims_buffer_size,
+    const void* optional_parameters,
+    size_t optional_parameters_size,
+    uint8_t** evidence_buffer,
+    size_t* evidence_buffer_size,
+    uint8_t** endorsements_buffer,
+    size_t* endorsements_buffer_size);
+
+typedef struct _oe_inittime_custom_claim_buffer_t
+{
+    /* ID of integrity algorithm to be used to verify the buffer content.
+     *   0: config_id = SHA256(buffer[]);
+     *   others: undefined;
+     */
+    uint32_t integrity_algorithm_id;
+    /* Variable length buffer */
+    char buffer[];
+} oe_inittime_custom_claim_buffer_t;
+```
+
```

Comment:
I tried to write sample code for the proposed API, just to make sure my understanding is correct:

```
    // Set up extra content and config id
    const char extra_content[] = "1234567890";
    oe_sgx_config_data_t = { SHA256(extra_content) /* config_id */, 0 /* config_svn */ };
    oe_enclave_setting_t settings[] =
    {
        {
            .setting_type = OE_ENCLAVE_INITTIME_CONFIGURATION_DATA,
            .u.oe_sgx_config_data_t = &oe_sgx_config_data_t
        }
    };

    // Create enclave
    oe_create_enclave(..., settings, ...);

    // Set up inittime claims
    oe_inittime_custom_claim_buffer_t inittime_custom_claims_buffer =
    {
        0, //  config_id = SHA256(buffer[])
        extra_content
    };
    size_t inittime_custom_claims_buffer_size =
        sizeof(oe_inittime_custom_claim_buffer_t) +
        sizeof(extra_content);

    // Generate evidence #1
    oe_get_evidence(...
        runtime_custom_claims_buffer_1,
        runtime_custom_claims_buffer_size_1,
        &inittime_custom_claims_buffer,
        inittime_custom_claims_buffer_size,
        ...);

    // Generate evidence #2
    oe_get_evidence(...
        runtime_custom_claims_buffer_2,
        runtime_custom_claims_buffer_size_2,
        &inittime_custom_claims_buffer,
        inittime_custom_claims_buffer_size,
        ...);

```

Reply:
pseudocode now included in the latest version of the PR.